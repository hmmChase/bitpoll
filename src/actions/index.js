export const storeUser = user => ({
  type: 'STORE_USER',
  user
});

export const storeLogOut = () => ({
  type: 'STORE_LOG_OUT'
});

export const storeContributors = contributors => ({
  type: 'STORE_CONTRIBUTORS',
  contributors
});

export const getContributors = url => ({
  type: 'GET_CONTRIBUTORS',
  url
});

export const storeIsContributor = boolean => ({
  type: 'STORE_IS_CONTRIBUTOR',
  boolean
});

export const storeIfVoted = boolean => ({
  type: 'STORE_IF_VOTED',
  boolean
});

export const storeSelectedOption = option => ({
  type: 'STORE_SELECTED_OPTION',
  option
});

export const storeOptionValue = value => ({
  type: 'STORE_OPTION_VALUE',
  value
});

export const storeVoteBtnDisabled = boolean => ({
  type: 'STORE_VOTE_BTN_DISABLED',
  boolean
});

export const storeOption1Tally = tally => ({
  type: 'STORE_OPTION1_TALLY',
  tally
});

export const storeOption2Tally = tally => ({
  type: 'STORE_OPTION2_TALLY',
  tally
});
