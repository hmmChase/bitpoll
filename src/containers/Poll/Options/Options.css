ul {
  padding: 0;
  list-style-type: none;
  margin-bottom: 20px;
}

ul li input[type='radio'] {
  visibility: hidden;
}

ul li label {
  cursor: pointer;
  margin-left: 5px;
  font-size: 1.5em;
}

ul li .check {
  display: inline-block;
  border: 5px solid #aaaaaa;
  border-radius: 100%;
  height: 25px;
  width: 25px;
  transition: border 0.25s linear;
  -webkit-transition: border 0.25s linear;
}

ul li .check::before {
  transition: background 0.25s linear;
  -webkit-transition: background 0.25s linear;
}

input[type='radio']:checked ~ .check {
  border: 10px solid #77e29c;
}

input[type='radio']:checked ~ .check::before {
  background: #77e29c;
}
