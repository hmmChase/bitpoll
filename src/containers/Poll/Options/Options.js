import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../../actions';
import './Options.css';

export class Options extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: ''
    };
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({
      value
    });
    this.props.storeSelectedOption(name);
    this.props.storeOptionValue(value);
    this.props.storeVoteBtnDisabled(false);
  };

  render() {
    return (
      <div>
        <ul>
          <li>
            <input
              className="option1"
              id="option1"
              name="option1"
              type="radio"
              value="yes"
              checked={this.state.value === 'yes'}
              onChange={this.handleChange}
            />
            <div className="check" />
            <label htmlFor="option1">Yes</label>
          </li>
          <li>
            <input
              id="option2"
              name="option2"
              type="radio"
              value="no"
              checked={this.state.value === 'no'}
              onChange={this.handleChange}
            />
            <div className="check" />
            <label htmlFor="option2">No</label>
          </li>
        </ul>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  storeSelectedOption: option => dispatch(actions.storeSelectedOption(option)),
  storeOptionValue: value => dispatch(actions.storeOptionValue(value)),
  storeVoteBtnDisabled: boolean =>
    dispatch(actions.storeVoteBtnDisabled(boolean))
});

export default connect(
  null,
  mapDispatchToProps
)(Options);
