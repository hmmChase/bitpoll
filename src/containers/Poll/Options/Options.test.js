import React from 'react';
import { shallow } from 'enzyme';
import { Options, mapDispatchToProps } from './Options';
import * as actions from '../../../actions';

describe('Options', () => {
  let options;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      storeSelectedOption: jest.fn(),
      storeOptionValue: jest.fn(),
      storeVoteBtnDisabled: jest.fn()
    };
    options = shallow(<Options {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(options).toMatchSnapshot();
  });

  it.skip('calls handleChange onChange', () => {
    const mockEvent = { target: { name: 'option1', value: 'yes' } };
    const handleChange = (options.instance().handleChange = jest.fn());
    options.find('.option1').simulate('change', mockEvent);

    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  describe('handleChange', () => {
    it('sets state.value on check of radio button', () => {
      const mockEvent = {
        target: {
          name: 'option1',
          value: 'yes'
        }
      };
      options.instance().handleChange(mockEvent);

      expect(options.state()).toEqual({
        value: 'yes'
      });
    });

    it('calls storeSelectedOption with correct params', () => {
      const mockEvent = {
        target: {
          name: 'option1',
          value: 'yes'
        }
      };
      options.instance().handleChange(mockEvent);

      expect(mockProps.storeSelectedOption).toHaveBeenCalledTimes(1);
      expect(mockProps.storeSelectedOption).toHaveBeenCalledWith('option1');
    });

    it('calls storeOptionValue with correct params', () => {
      const mockEvent = {
        target: {
          name: 'option1',
          value: 'yes'
        }
      };
      options.instance().handleChange(mockEvent);

      expect(mockProps.storeOptionValue).toHaveBeenCalledTimes(1);
      expect(mockProps.storeOptionValue).toHaveBeenCalledWith('yes');
    });

    it('calls storeVoteBtnDisabled with correct params', () => {
      const mockEvent = {
        target: {
          name: 'option1',
          value: 'yes'
        }
      };
      options.instance().handleChange(mockEvent);

      expect(mockProps.storeVoteBtnDisabled).toHaveBeenCalledTimes(1);
      expect(mockProps.storeVoteBtnDisabled).toHaveBeenCalledWith(false);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch on storeSelectedOption', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.storeSelectedOption();
      mappedProps.storeSelectedOption();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch on storeOptionValue', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.storeOptionValue();
      mappedProps.storeOptionValue();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch on storeVoteBtnDisabled', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.storeVoteBtnDisabled();
      mappedProps.storeVoteBtnDisabled();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});
