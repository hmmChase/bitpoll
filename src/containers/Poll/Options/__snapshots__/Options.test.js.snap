// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Options
    storeOptionValue={[Function]}
    storeSelectedOption={[Function]}
    storeVoteBtnDisabled={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                <input
                        checked={false}
                        className="option1"
                        id="option1"
                        name="option1"
                        onChange={[Function]}
                        type="radio"
                        value="yes"
                />
                <div
                        className="check"
                />
                <label
                        htmlFor="option1"
                >
                        Yes
                </label>
        </li>
        <li>
                <input
                        checked={false}
                        id="option2"
                        name="option2"
                        onChange={[Function]}
                        type="radio"
                        value="no"
                />
                <div
                        className="check"
                />
                <label
                        htmlFor="option2"
                >
                        No
                </label>
        </li>
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <input
                        checked={false}
                        className="option1"
                        id="option1"
                        name="option1"
                        onChange={[Function]}
                        type="radio"
                        value="yes"
            />
            <div
                        className="check"
            />
            <label
                        htmlFor="option1"
            >
                        Yes
            </label>
</li>,
          <li>
            <input
                        checked={false}
                        id="option2"
                        name="option2"
                        onChange={[Function]}
                        type="radio"
                        value="no"
            />
            <div
                        className="check"
            />
            <label
                        htmlFor="option2"
            >
                        No
            </label>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className="option1"
                id="option1"
                name="option1"
                onChange={[Function]}
                type="radio"
                value="yes"
/>,
              <div
                className="check"
/>,
              <label
                htmlFor="option1"
>
                Yes
</label>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": "option1",
                "id": "option1",
                "name": "option1",
                "onChange": [Function],
                "type": "radio",
                "value": "yes",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "check",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "htmlFor": "option1",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "label",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="option2"
                name="option2"
                onChange={[Function]}
                type="radio"
                value="no"
/>,
              <div
                className="check"
/>,
              <label
                htmlFor="option2"
>
                No
</label>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "option2",
                "name": "option2",
                "onChange": [Function],
                "type": "radio",
                "value": "no",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "check",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "htmlFor": "option2",
              },
              "ref": null,
              "rendered": "No",
              "type": "label",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                    <input
                              checked={false}
                              className="option1"
                              id="option1"
                              name="option1"
                              onChange={[Function]}
                              type="radio"
                              value="yes"
                    />
                    <div
                              className="check"
                    />
                    <label
                              htmlFor="option1"
                    >
                              Yes
                    </label>
          </li>
          <li>
                    <input
                              checked={false}
                              id="option2"
                              name="option2"
                              onChange={[Function]}
                              type="radio"
                              value="no"
                    />
                    <div
                              className="check"
                    />
                    <label
                              htmlFor="option2"
                    >
                              No
                    </label>
          </li>
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <input
                            checked={false}
                            className="option1"
                            id="option1"
                            name="option1"
                            onChange={[Function]}
                            type="radio"
                            value="yes"
              />
              <div
                            className="check"
              />
              <label
                            htmlFor="option1"
              >
                            Yes
              </label>
</li>,
            <li>
              <input
                            checked={false}
                            id="option2"
                            name="option2"
                            onChange={[Function]}
                            type="radio"
                            value="no"
              />
              <div
                            className="check"
              />
              <label
                            htmlFor="option2"
              >
                            No
              </label>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="option1"
                  id="option1"
                  name="option1"
                  onChange={[Function]}
                  type="radio"
                  value="yes"
/>,
                <div
                  className="check"
/>,
                <label
                  htmlFor="option1"
>
                  Yes
</label>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "option1",
                  "id": "option1",
                  "name": "option1",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "yes",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "check",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Yes",
                  "htmlFor": "option1",
                },
                "ref": null,
                "rendered": "Yes",
                "type": "label",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  id="option2"
                  name="option2"
                  onChange={[Function]}
                  type="radio"
                  value="no"
/>,
                <div
                  className="check"
/>,
                <label
                  htmlFor="option2"
>
                  No
</label>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "id": "option2",
                  "name": "option2",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "no",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "check",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No",
                  "htmlFor": "option2",
                },
                "ref": null,
                "rendered": "No",
                "type": "label",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
